File: CourseRegistrationSystem/src/Domain/StaffIDGenerator.java
package Domain;

import java.util.ArrayList;
import java.util.Collection;
import java.util.List;

public class StaffIDGenerator extends IDGenerator {
        
    public StaffIDGenerator(){
        super();
        this.setDigitCount(8);
        this.setIdPrefix("100");
        this.setIdSuffix("");
        this.setUsedIDs(getUsedIDsByOtherStaffs());
    }

    private List<String> getUsedIDsByOtherStaffs(){

        Collection<Staff> allStaff =  Department.getAllStaffs();
        List<String> alreadyUsedIDs = new ArrayList<>();

        for (Staff staff : allStaff) {
            alreadyUsedIDs.add(staff.getStaffNo());
        }
        return alreadyUsedIDs;
    }
}

File: Iteration3/IDGenerators/StaffIDGenerator.py
from IDGenerators.IDGenerator import IDGenerator
import random

class StaffIDGenerator(IDGenerator):
    def __init__(self):
        super().__init__()
        self.set_digit_count(8)
        self.set_id_prefix("100")
        self.set_id_suffix("")
        self.set_used_ids(['']) # (self.get_used_ids_by_other_staffs())


    def get_used_ids_by_other_staffs(self):
        from Domain.Department import Department
        all_staff = Department.get_instance().get_all_staffs()
        already_used_ids = [staff.get_staff_no() for staff in all_staff]
        return already_used_ids

